tsfresh/feature_extraction/data.py

```diff
@@ -0,0 +1,371 @@
+import itertools
+from typing import Iterable
+
+import pandas as pd
+
+
+class TsData(Iterable):
+    """
+    TsData provides access to time series data for internal usage.
+
+    Implementations must at least overwrite `__iter__` which must yield tuples containing
+    (id, kind, pd.Series). Make sure `kind` is of type `str` to allow inference
+    of feature settings in `feature_extraction.settings.from_columns`.
+
+    Other methods should be overwritten if a more efficient solution exists for the underlying data store.
+    """
+
+    def __iter__(self):
+        raise NotImplementedError
+
+    def __len__(self):
+        return sum(1 for _ in self)
+
+    def partition(self, chunk_size):
+        """
+        Split the data into iterable chunks of given `chunk_size`.
+
+        :param chunk_size: the size of the chunks
+        :type chunk_size int
+
+        :return: Generator of iterables
+        """
```

Comment:
actually, the partition method is a generator, so I think the docstring should be 

```suggestion
        :return: Generator of iterables
        """
```
```suggestion
        :yields: chunks of values 
        """
```

Reply:
Sounds good. Didn't knew `:yields:` exists.

Reply:
It seems PyCharm has no support for this directive. Also I, can't find any documentation for this, nor is it already used anywhere inside tsfresh. Unless you insist on that, I suggest that we stick to `:return`